src/modules/clusters/cluster-list/cluster-list.js

```diff
@@ -100,6 +96,326 @@
+                    placeholder: "Filter by Name",
+                    filterType: "text"
+                }, {
+                    id: "ipAddress",
+                    title: "Address",
+                    placeholder: "Filter by IP Address",
+                    filterType: "text"
+                }],
+                appliedHostFilters: [],
+                onFilterChange: _filterHostChange,
+            };
+
+            vm.hostCluster.tableConfig = {
+                selectionMatchProp: "fqdn",
+                itemsAvailable: true,
+                showCheckboxes: false
+            };
+
+            vm.hostCluster.tableColumns = [{
+                header: "Host",
+                itemField: "fqdn"
+            }, {
+                header: "Address",
+                itemField: "ipAddress"
+            }];
+
+            vm.hostCluster.filterConfig.resultsCount = vm.hostCluster.filtered.length;
+
+        }
+
+        function closeHostModal(dismissCause) {
+            vm.showHost = false;
+        }
+
+        //***Private Functions***
+
+        function _matchesHostFilter(item, filter) {
+            var match = true;
+            var re = new RegExp(filter.value, "i");
+
+            if (filter.id === "fqdn") {
+                match = item.fqdn.match(re) !== null;
+            } else if (filter.id === "ipAddress") {
+                match = item.ipAddress.match(re) !== null;
+            }
+            return match;
+        }
+
+        function _matchesHostFilters(item, filters) {
+            var matches = true;
+
+            filters.forEach(function(filter) {
+                if (!_matchesHostFilter(item, filter)) {
+                    matches = false;
+                    return false;
+                }
+            });
+            return matches;
+        }
+
+        function _applyHostFilters(filters) {
+            vm.hostCluster.filtered = [];
+            if (filters && filters.length > 0) {
+                vm.hostCluster.hosts.forEach(function(item) {
+                    if (_matchesHostFilters(item, filters)) {
+                        vm.hostCluster.filtered.push(item);
+                    }
+                });
+            } else {
+                vm.hostCluster.filtered = vm.hostCluster.hosts;
+            }
+            vm.hostCluster.filterConfig.resultsCount = vm.hostCluster.filtered.length;
+        }
+
+        function _filterHostChange(filters) {
+            vm.hostCluster.filtersText = "";
+            vm.hostCluster.filters = filters;
+            filters.forEach(function(filter) {
+                vm.hostCluster.filtersText += filter.title + " : ";
+                if (filter.value.title) {
+                    vm.hostCluster.filtersText += filter.value.title;
+                } else {
+                    vm.hostCluster.filtersText += filter.value;
+                }
+                vm.hostCluster.filtersText += "\n";
+            });
+            _applyHostFilters(filters);
+        }
+
```

Comment:
@a2batic `_filterHostChange`,  `_applyHostFilters` and `_matchesHostFilters` can be made generic for all the modals used in this view instead of creating separate one for each.